lib/game.rb

```diff
@@ -1,16 +1,60 @@
 require './lib/ship'
 require './lib/cell'
 require './lib/board'
+require './lib/turn'
+
 
 class Game
-  attr_reader :user_board, :comp_board
+
+  attr_reader :user_board, :comp_board, :previous_shots, :user_cruiser, :user_sub
 
   def initialize
     @comp_board = Board.new
     @user_board = Board.new
+    @user_previous_shots = []
+    @cpu_previous_shots =[]
+
     # require "pry"; binding.pry
   end
 
+  def user_take_turn
+    puts "Choose a coordinate upon which to fire."
+    fire = gets.chomp
+
+    unless @comp_board.cells.include?(fire)
+      puts "Please input a valid coordinate."
+      fire = gets.chomp
+    end
+
+    turn = Turn.new(fire, @comp_board)
+
+    while @user_previous_shots.include?(fire)
+      puts "That cell has already been shot. Please choose another."
+      fire = gets.chomp
+      turn = Turn.new(fire, @comp_board)
+    end
+  turn.take_shot
```

Comment:
Does the user's previous shots refer to the ones the user has shot or the ones on his board that have been shot?